### ISSUE TITLE
RFE: execute rpmbuild tests as a regular user

### ISSUE THREAD

**Issue by pmatilai** [rpm-software-management/rpm#3005]
While investigating #2519 I realized that we're running the entire test-suite as root. That's not how rpmbuild is intended to be used, and masks various permission issues real-world users have, such as that #2519 which is not reproducable in the test-suite because of this.

Adding a user to the test-environment is trivial enough, actually running the tests with it probably less so. Besides all rpmbuild tests, there's probably a lot of other tests too that would optimally run as non-root.

**Comment by dmnks** [comment]
Yup, good point. I think we should actually make the `rpmtests` script (which runs in the podman container) run as a regular user there. The individual tests aren't supposed to write to the root filesystem anyway (which we prevent by making it read-only) so being root shouldn't be necessary either. Typically, the tests would run `rpm` to query something and/or prepare the `$PWD` which can be done under a regular user.

Then, when a test actually needs to write to the system (and thus needs root privileges), we can escalate it (via `sudo` perhaps) in the `runroot*` functions. This is also how Toolbox works - the user in the container is mapped to your native user on the host and you have to use `sudo` in the same way as you'd on the host.

**Comment by dmnks** [comment]
Bonus point - `runroot` will finally live up to its name :smile:

**Comment by pmatilai** [comment]
Sounds like a plan :+1:

**Comment by xsuchy** [comment]
> we're running the entire test-suite as root.

I believe this is not true. I see no code in rpmbuild that would elevate UID to root. Nor any consolehelper. Nor setuid bits.

**Comment by dmnks** [comment]
What I mean is rpm's own test-suite: https://github.com/rpm-software-management/rpm/blob/5d4a476d14998f8f7ebc7e0c15a5263ca7803f5d/tests/mktree.oci#L53

**Comment by pmatilai** [comment]
> I believe this is not true. I see no code in rpmbuild that would elevate UID to root. Nor any consolehelper. Nor setuid bits.

In the container.
